File: src/main/java/sk/tuke/gamestudio/game/BlockPuzzle/levels/EasyLevel.java
package sk.tuke.gamestudio.game.BlockPuzzle.levels;

import sk.tuke.gamestudio.game.BlockPuzzle.consoleui.ConsoleColor;
import sk.tuke.gamestudio.game.BlockPuzzle.core.board.Cell;
import sk.tuke.gamestudio.game.BlockPuzzle.core.board.Block;
import sk.tuke.gamestudio.game.BlockPuzzle.core.board.Board;

import java.util.ArrayList;
import java.util.List;

public class EasyLevel implements Level{
    @Override
    public Board generateBoard() {
        return new Board(GameLevels.EASY.getRows(), GameLevels.EASY.getCols());
    }

    @Override
    public List<Block> generateBlocks() {
        List<Block> blocks = new ArrayList<>();

        blocks.add(new Block( new Cell[][]{
                {new Cell(ConsoleColor.RED), new Cell(ConsoleColor.RED), new Cell(ConsoleColor.RED), new Cell(ConsoleColor.RED)},   // ####
                {new Cell(ConsoleColor.RED), Cell.createEmptyBlockCell(), Cell.createEmptyBlockCell(), Cell.createEmptyBlockCell()} // #
        }, ConsoleColor.RED));
        blocks.add(new Block( new Cell[][]{
                {Cell.createEmptyBlockCell(), new Cell(ConsoleColor.BLUE), new Cell(ConsoleColor.BLUE)}, //  ##
                {new Cell(ConsoleColor.BLUE), new Cell(ConsoleColor.BLUE), Cell.createEmptyBlockCell()}  // ##
        }, ConsoleColor.BLUE));
        blocks.add(new Block( new Cell[][]{
                {new Cell(ConsoleColor.GREEN), new Cell(ConsoleColor.GREEN)}, // ##
                {new Cell(ConsoleColor.GREEN), new Cell(ConsoleColor.GREEN)}  // ##
        }, ConsoleColor.GREEN));
        blocks.add(new Block( new Cell[][]{
                {Cell.createEmptyBlockCell(), new Cell(ConsoleColor.CYAN)},      //  #
                {new Cell(ConsoleColor.CYAN), new Cell(ConsoleColor.CYAN)},      // ##
                {new Cell(ConsoleColor.CYAN), Cell.createEmptyBlockCell()}       // #
        }, ConsoleColor.CYAN));
        blocks.add(new Block( new Cell[][]{
                {new Cell(ConsoleColor.YELLOW), Cell.createEmptyBlockCell()},   // #
                {new Cell(ConsoleColor.YELLOW), new Cell(ConsoleColor.YELLOW)}, // ##
                {new Cell(ConsoleColor.YELLOW), new Cell(ConsoleColor.YELLOW)}  // ##
        }, ConsoleColor.YELLOW));
        blocks.add(new Block( new Cell[][]{
                {new Cell(ConsoleColor.PURPLE)}, // #
                {new Cell(ConsoleColor.PURPLE)}, // #
                {new Cell(ConsoleColor.PURPLE)}  // #
        }, ConsoleColor.PURPLE));

        return blocks;
    }
}

File: gamestudio-frontend/src/core/level_generators/easyLevel.js
import Block from '../Block';
import Cell from '../cell/Cell';
import colors from '../cell/Colors';
import CellState from '../cell/CellState';

const generateEasyBlocks = () => {
    return [
        new Block(
            [
                [new Cell(CellState.FILLED, colors.RED), new Cell(CellState.FILLED, colors.RED), new Cell(CellState.FILLED, colors.RED), new Cell(CellState.FILLED, colors.RED)],
                [new Cell(CellState.FILLED, colors.RED), new Cell(CellState.EMPTY), new Cell(CellState.EMPTY), new Cell(CellState.EMPTY)],
            ],
            colors.RED
        ),
        new Block(
            [
                [new Cell(CellState.EMPTY), new Cell(CellState.FILLED, colors.BLUE), new Cell(CellState.FILLED, colors.BLUE)],
                [new Cell(CellState.FILLED, colors.BLUE), new Cell(CellState.FILLED, colors.BLUE), new Cell(CellState.EMPTY)],
            ],
            colors.BLUE
        ),
        new Block(
            [
                [new Cell(CellState.FILLED, colors.GREEN), new Cell(CellState.FILLED, colors.GREEN)],
                [new Cell(CellState.FILLED, colors.GREEN), new Cell(CellState.FILLED, colors.GREEN)],
            ],
            colors.GREEN
        ),
        new Block(
            [
                [new Cell(CellState.EMPTY), new Cell(CellState.FILLED, colors.CYAN)],
                [new Cell(CellState.FILLED, colors.CYAN), new Cell(CellState.FILLED, colors.CYAN)],
                [new Cell(CellState.FILLED, colors.CYAN), new Cell(CellState.EMPTY)],
            ],
            colors.CYAN
        ),
        new Block(
            [
                [new Cell(CellState.FILLED, colors.YELLOW), new Cell(CellState.EMPTY)],
                [new Cell(CellState.FILLED, colors.YELLOW), new Cell(CellState.FILLED, colors.YELLOW)],
                [new Cell(CellState.FILLED, colors.YELLOW), new Cell(CellState.FILLED, colors.YELLOW)],
            ],
            colors.YELLOW
        ),
        new Block(
            [
                [new Cell(CellState.FILLED, colors.PURPLE)],
                [new Cell(CellState.FILLED, colors.PURPLE)],
                [new Cell(CellState.FILLED, colors.PURPLE)],
            ],
            colors.PURPLE
        ),
    ];
};

export default generateEasyBlocks;